joomla/joomla-cms
# Allow onValidateContact plugin to show error message

Pull Request for Issue #37584 .

### Summary of Changes
This pull request copies actions performed on contact form submissions that fail native form validation, and applies it to submissions that fail `onValidateContact` plugins.



### Testing Instructions
This [dummy plugin](https://www.dropbox.com/s/v2x130bl10mbqsm/plg_contact_dummyvalidation.zip?dl=0) always fail contact form submissions and should demonstrate the issue:
```
<?php
defined('_JEXEC') or die;

class plgContactDummyValidation extends JPlugin {

	protected $autoloadLanguage = true;

	function onValidateContact($contact, $data) {
		return new InvalidArgumentException("Exception message is not shown.");
	}
}
```

1. Install dummy plugin above.
(System > Manage > Extensions > Install Extensions > Upload Package File)
2. Enable dummy plugin.
(System > Manage > Plugins > Contact - Dummy Validation > Enable plugin)
3. Create a contact form in admin panel.
(Menus > Main Menu > Add Site Menu Item > Menu Item Type > Contacts > Single Contact > Select Contact [create if there isn't already one], and give it a title)
4. Set up valid mail credentials if not already present.
(System > Setup > Global Configuration > Server > Mail)
5. Fill and submit contact form on the public site. Observe how user is redirected to a blank page.
6. Apply patch.
7. Fill and submit contact form again. Observe how user is redirected to contact form pre-filled with previously entered values, with exception message in an alert banner which plugins can use to explain why the submission was unsuccessful.

### Actual result BEFORE applying this Pull Request
If user submits a contact form with parameters failing an `onValidateContact` triggered plugin, plugin returns an exception, and user is redirected to a blank contact form page, stripped of all form elements with no error message indicating why the submission was unsuccessful.
![Screenshot from 2022-04-12 23-41-00](https://user-images.githubusercontent.com/103491197/163001047-f4673c35-2c1e-42e2-8a1e-6301bbdaca1a.png)



### Expected result AFTER applying this Pull Request
User is redirected to contact form page, pre-filled with submitted values and displaying exception message explanation of why submission failed. Same as if submission had failed native form validation.
![Screenshot from 2022-04-21 11-42-00](https://user-images.githubusercontent.com/103491197/164367612-a4833049-e77a-4506-b919-fbefda407634.png)



### Documentation Changes Required
None.
